JFRWebSocket.m

```diff
@@ -717,6 +725,46 @@ -(void)dealloc
 
 /////////////////////////////////////////////////////////////////////////////
 /////////////////////////////////////////////////////////////////////////////
+@interface JFRThread ()
+
+@property(nonatomic, assign)BOOL isRunning;
+@property(nonatomic, strong)NSRunLoop *loop;
+
+@end
+/////////////////////////////////////////////////////////////////////////////
+@implementation JFRThread
+
+/////////////////////////////////////////////////////////////////////////////
++(NSRunLoop*)sharedLoop {
+    return [self shared].loop;
+}
+/////////////////////////////////////////////////////////////////////////////
++(JFRThread*)shared {
+    static JFRThread *manager = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        manager = [[[self class] alloc] init];
```

Comment:
In a class method, `self` is already a `Class`, so this is redundant:
`manager = [[[self class] alloc] init];`
Better as:
`manager = [[self alloc] init];`